src/aish/llm/session.py

```
@@ -1775,7 +1775,13 @@ async def process_input(
                             msg = response["choices"][0]["message"]  # type: ignore
                             finish_reason = response["choices"][0]["finish_reason"]  # type: ignore
                 except TimeoutError:
-                    events.emit_cancelled("llm_timeout")
+                    if (
+                        self.cancellation_token
+                        and self.cancellation_token.is_cancelled()
+                    ):
+                        events.emit_cancelled("llm_cancelled")
+                        events.emit_generation_end(status="cancelled")
+                        raise anyio.get_cancelled_exc_class()
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)